src/ol/geom/linestring.js

```diff
@@ -246,7 +246,7 @@ ol.geom.LineString.prototype.intersectsExtent = function(extent) {
  */
 ol.geom.LineString.prototype.setCoordinates = function(coordinates, opt_layout) {
   if (!coordinates) {
-    this.setFlatCoordinates(ol.geom.GeometryLayout.XY, null);
+    this.setFlatCoordinates(ol.geom.GeometryLayout.XY, []);
```

Comment:
Yes. The EMPTY tests does not pass otherwise.
Could be handled in `getCoordinates` instead. That code does not handle that this.flatCoordinates is null and throws.